docs/cbuild-run-enhancement.md

```diff
@@ -0,0 +1,129 @@
+&nbsp;&nbsp;&nbsp; _`punits:`_                    |        |   Optional   | Specifies processor units in a symmetric multi-processor core (MPCore). Mandatory if multiple CPU debug blocks are accessible via a single AP.
+&nbsp;&nbsp;&nbsp; `apid:`                        | uint   | **Required** | Default access port ID to use for target accesses in this debug connection.
+&nbsp;&nbsp;&nbsp; `defaultResetSequence:`        | string |   Optional   | Specifies the debug sequence that shall be used by default for the reset operation. If not specified then `ResetSystem` sequence will be set as default.
+
+_`punits:`_                                       | Value  |   Use        | Content
+:-------------------------------------------------|--------|--------------|:------------------------------------
+_`- punit:`_                                      | uint   | **Required** | Specifies a specific processor unit of a symmetric MPCore.
+&nbsp;&nbsp;&nbsp; _`address:`_                   | uint   | **Required** | Specifies the base address of the CPU debug block.
+
+>Nodes in _italic_ can be skipped for now and considered in the future.
+
+### `programming:`
+`programming:`                                    | Value  |   Use        | Content
+:-------------------------------------------------|--------|--------------|:------------------------------------
+`- algorithm:`                                    | string | **Required** | Flash Programming Algorithm file including the path.
+&nbsp;&nbsp;&nbsp; `start:`                       | uint   | **Required** | Start address of flash covered by the Flash programming algorithm.
+&nbsp;&nbsp;&nbsp; `size:`                        | uint   | **Required** | Size of flash covered by the Flash programming algorithm.
+&nbsp;&nbsp;&nbsp; `ram-start:`                   | uint   | **Required** | Start address of the RAM where the Flash programming algorithm will be executed from.
+&nbsp;&nbsp;&nbsp; `ram-size:`                    | uint   | **Required** | Maximum size of RAM available for the execution of the Flash programming algorithm.
+&nbsp;&nbsp;&nbsp; `default:`                     | bool   |   Optional   | If `true`, then this is the default Flash programming algorithm that gets configured in a project. Default value is `false`.
+&nbsp;&nbsp;&nbsp; `pname:`                       | string |   Optional   | Processor identifier. Mandatory for devices that embed multiple processors that require different algorithms.
+
+
+### `system-descriptions:`
+`system-descriptions:`                            | Value  |   Use        | Content
+:-------------------------------------------------|--------|--------------|:------------------------------------
+`- file:`                                         | string | **Required** | Specifies the file name including the path.
+&nbsp;&nbsp;&nbsp; `type:`                        | string | **Required** | Specifies the file type. Predefined values: `svd`, `sdf`.

```

Comment:
We need to think about where to put `scvd` and other files that are strictly speaking not system descriptions and, IIRC, not even CMSIS description files. Yet such additional debug support files may want to find their place in the `cbuild-run` files. Maybe as part of the `debugger` node?

Reply:
Maybe we introduce `component-descriptions:` node for scvd. We could also think about a more common node than `system-descriptions:` and use it for `svd`, `sdf`, `scvd`.